modules/programs/zsh.nix

```diff
@@ -186,6 +186,24 @@ in
         description = "Extra commands that should be added to <filename>.zshrc</filename>.";
       };
 
+      profileExtra = mkOption {
+        default = null;
+        type = types.nullOr types.lines;
```

Comment:
The default value can be an empty string. In that case, you can remove conditions below.

Reply:
The type can be `types.lines` and the condition `cfg.profileExtra != ""`, then it won't try to write an empty file by default, which will prevent a conflict with preexisting such files.

Reply:
Good point @Infinisil, haven't thought about it.

Reply:
@Infinisil Ah, smart! Didn't want to generate files, which is why I kept `nullOr`, but that is better.